File: arb-www/src/schema/nfl/standings.ts
import { BaseDTO } from '../BaseDTO';

/**
 * NFL Standings information
 */
export class NFLStanding extends BaseDTO {
  SeasonType!: number;
  Season!: number;
  Conference!: string;
  Division!: string;
  Team!: string;
  Name!: string;
  Wins!: number;
  Losses!: number;
  Ties!: number;
  Percentage!: number;
  PointsFor!: number;
  PointsAgainst!: number;
  NetPoints!: number;
  Touchdowns!: number;
  DivisionWins!: number;
  DivisionLosses!: number;
  ConferenceWins!: number;
  ConferenceLosses!: number;
  TeamID!: number;
  DivisionTies!: number;
  ConferenceTies!: number;
  GlobalTeamID!: number;
  DivisionRank!: number;
  ConferenceRank!: number;
  HomeWins!: number;
  HomeLosses!: number;
  HomeTies!: number;
  AwayWins!: number;
  AwayLosses!: number;
  AwayTies!: number;
  Streak!: number;
}

/**
 * NFL Standings response
 */
export class NFLStandingsResponse extends BaseDTO {
  data!: NFLStanding[];
}

File: arb-rs/src/schema/nfl/standings.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "PascalCase")]
pub struct NFLStanding {
    pub season_type: i32,
    pub season: i32,
    pub conference: String,
    pub division: String,
    pub team: String,
    pub name: String,
    pub wins: i32,
    pub losses: i32,
    pub ties: i32,
    pub percentage: f64,
    pub points_for: i32,
    pub points_against: i32,
    pub net_points: i32,
    pub touchdowns: i32,
    pub division_wins: i32,
    pub division_losses: i32,
    pub conference_wins: i32,
    pub conference_losses: i32,
    #[serde(rename = "TeamID")]
    pub team_id: i32,
    pub division_ties: i32,
    pub conference_ties: i32,
    #[serde(rename = "GlobalTeamID")]
    pub global_team_id: i32,
    pub division_rank: i32,
    pub conference_rank: i32,
    pub home_wins: i32,
    pub home_losses: i32,
    pub home_ties: i32,
    pub away_wins: i32,
    pub away_losses: i32,
    pub away_ties: i32,
    pub streak: i32,
}

pub type NFLStandings = Vec<NFLStanding>;
